### ISSUE TITLE
ch5 rgbd jointMap.cpp

### ISSUE THREAD

**Issue by iamkrs9** [gaoxiang12/slambook2#167]
I had a problem making jointMap with cmake.

**TL;DR:
Solved it with the help of this:
https://github.com/fmtlib/fmt/issues/596#issuecomment-341617817

Specifically, I just put the following line in jointMap.cpp
#define FMT_HEADER_ONLY**

It gave the following error:

Undefined symbols for architecture x86_64:
  "fmt::v7::format_error::~format_error()", referenced from:
      fmt::v7::detail::fill_t<char>::operator=(fmt::v7::basic_string_view<char>) in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_error() in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_error() in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned __int128>::on_error() in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, float, 0>(fmt::v7::detail::buffer_appender<char>, float, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, double, 0>(fmt::v7::detail::buffer_appender<char>, double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, long double, 0>(fmt::v7::detail::buffer_appender<char>, long double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      ...
  "fmt::v7::detail::basic_data<void>::hex_digits", referenced from:
      char* fmt::v7::detail::format_uint<4u, char, unsigned int>(char*, unsigned int, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<1u, char, unsigned int>(char*, unsigned int, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<3u, char, unsigned int>(char*, unsigned int, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<4u, char, unsigned long long>(char*, unsigned long long, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<1u, char, unsigned long long>(char*, unsigned long long, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<3u, char, unsigned long long>(char*, unsigned long long, int, bool) in joinMap.cpp.o
      char* fmt::v7::detail::format_uint<4u, char, unsigned __int128>(char*, unsigned __int128, int, bool) in joinMap.cpp.o
      ...
  "fmt::v7::detail::basic_data<void>::left_padding_shifts", referenced from:
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)1, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, char, fmt::v7::detail::buffer_appender<char> >(fmt::v7::detail::buffer_appender<char>, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&)::'lambda'(fmt::v7::detail::buffer_appender<char>)>(char, fmt::v7::basic_format_specs<fmt::v7::detail::buffer_appender<char> > const&, unsigned long, unsigned long, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, char, fmt::v7::detail::buffer_appender<char> >(fmt::v7::detail::buffer_appender<char>, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&)::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      __ZN3fmt2v76detail12write_paddedILNS0_5align4typeE1ENS1_15buffer_appenderIcEEcRZNS1_10write_charIcS6_EET0_S8_T_RKNS0_18basic_format_specsIS9_EEEUlS6_E_EES8_S8_RKNSA_IT1_EEmmOT2_ in joinMap.cpp.o
      __ZN3fmt2v76detail12write_paddedILNS0_5align4typeE1ENS1_15buffer_appenderIcEEcRZNS1_15write_nonfiniteIcS6_EET0_S8_bRKNS0_18basic_format_specsIT_EERKNS1_11float_specsEEUlS6_E_EES8_S8_RKNS9_IT1_EEmmOT2_ in joinMap.cpp.o
      __ZN3fmt2v76detail12write_paddedILNS0_5align4typeE1ENS1_15buffer_appenderIcEEcRZNS1_11write_bytesIcS6_EET0_S8_NS0_17basic_string_viewIcEERKNS0_18basic_format_specsIT_EEEUlS6_E_EES8_S8_RKNSB_IT1_EEmmOT2_ in joinMap.cpp.o
  "fmt::v7::detail::basic_data<void>::right_padding_shifts", referenced from:
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_dec()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned int)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned int> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_dec()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_hex()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned int)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned int> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_hex()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_bin()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned int)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned int> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_bin()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_oct()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned int)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned int> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_oct()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_num()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(char, fmt::v7::basic_format_specs<unsigned int> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_num()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_dec()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned long long)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned long long> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_dec()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      char fmt::v7::detail::write_padded<(fmt::v7::align::type)2, fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_int<fmt::v7::detail::buffer_appender<char>, char, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_hex()::'lambda'(fmt::v7::detail::buffer_appender<char>)>(fmt::v7::detail::buffer_appender<char>, int, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_specs<char> const&, unsigned long long)::'lambda'(fmt::v7::detail::buffer_appender<char>)&>(char, fmt::v7::basic_format_specs<unsigned long long> const&, unsigned long, unsigned long, fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_hex()::'lambda'(fmt::v7::detail::buffer_appender<char>)&&) in joinMap.cpp.o
      ...
  "fmt::v7::detail::basic_data<void>::zero_or_powers_of_10_32_new", referenced from:
      fmt::v7::detail::count_digits(unsigned int) in joinMap.cpp.o
  "fmt::v7::detail::basic_data<void>::zero_or_powers_of_10_64_new", referenced from:
      fmt::v7::detail::count_digits(unsigned long long) in joinMap.cpp.o
  "fmt::v7::detail::basic_data<void>::signs", referenced from:
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, float, 0>(fmt::v7::detail::buffer_appender<char>, float, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_nonfinite<char, fmt::v7::detail::buffer_appender<char> >(fmt::v7::detail::buffer_appender<char>, bool, fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::float_specs const&)::'lambda'(fmt::v7::detail::buffer_appender<char>)::operator()(fmt::v7::detail::buffer_appender<char>) const in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_float<fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp, char>(fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp const&, fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::float_specs, char)::'lambda'(fmt::v7::detail::buffer_appender<char>)::operator()(fmt::v7::detail::buffer_appender<char>) const in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_float<fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp, char>(fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp const&, fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::float_specs, char)::'lambda0'(fmt::v7::detail::buffer_appender<char>)::operator()(fmt::v7::detail::buffer_appender<char>) const in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_float<fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp, char>(fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp const&, fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::float_specs, char)::'lambda1'(fmt::v7::detail::buffer_appender<char>)::operator()(fmt::v7::detail::buffer_appender<char>) const in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_float<fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp, char>(fmt::v7::detail::buffer_appender<char>, fmt::v7::detail::big_decimal_fp const&, fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::float_specs, char)::'lambda2'(fmt::v7::detail::buffer_appender<char>)::operator()(fmt::v7::detail::buffer_appender<char>) const in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, double, 0>(fmt::v7::detail::buffer_appender<char>, double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      ...
  "fmt::v7::detail::basic_data<void>::digits", referenced from:
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned int>(char*, unsigned int, int) in joinMap.cpp.o
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned long long>(char*, unsigned long long, int) in joinMap.cpp.o
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned __int128>(char*, unsigned __int128, int) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write_exponent<char, fmt::v7::detail::buffer_appender<char> >(int, fmt::v7::detail::buffer_appender<char>) in joinMap.cpp.o
  "fmt::v7::detail::assert_fail(char const*, int, char const*)", referenced from:
      char const* fmt::v7::detail::parse_arg_id<char, fmt::v7::detail::id_adapter<fmt::v7::detail::format_string_checker<char, fmt::v7::detail::error_handler, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > >&, char>&>(char const*, char const*, fmt::v7::detail::id_adapter<fmt::v7::detail::format_string_checker<char, fmt::v7::detail::error_handler, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > >&, char>&) in joinMap.cpp.o
      int fmt::v7::detail::parse_nonnegative_int<char, fmt::v7::detail::id_adapter<fmt::v7::detail::format_string_checker<char, fmt::v7::detail::error_handler, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > >&, char>&>(char const*&, char const*, fmt::v7::detail::id_adapter<fmt::v7::detail::format_string_checker<char, fmt::v7::detail::error_handler, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > >&, char>&) in joinMap.cpp.o
      std::__1::make_unsigned<long>::type fmt::v7::detail::to_unsigned<long>(long) in joinMap.cpp.o
      decltype(fp.begin()) fmt::v7::formatter<fmt::v7::basic_string_view<char>, char, void>::parse<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> >(fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler>&) in joinMap.cpp.o
      char const* fmt::v7::detail::parse_align<char, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&>(char const*, char const*, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&) in joinMap.cpp.o
      char const* fmt::v7::detail::parse_width<char, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&>(char const*, char const*, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&) in joinMap.cpp.o
      int fmt::v7::detail::parse_nonnegative_int<char, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&>(char const*&, char const*, fmt::v7::detail::specs_checker<fmt::v7::detail::dynamic_specs_handler<fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler> > >&) in joinMap.cpp.o
      ...
  "int fmt::v7::detail::format_float<double>(double, int, fmt::v7::detail::float_specs, fmt::v7::detail::buffer<char>&)", referenced from:
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, float, 0>(fmt::v7::detail::buffer_appender<char>, float, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, double, 0>(fmt::v7::detail::buffer_appender<char>, double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "int fmt::v7::detail::format_float<long double>(long double, int, fmt::v7::detail::float_specs, fmt::v7::detail::buffer<char>&)", referenced from:
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, long double, 0>(fmt::v7::detail::buffer_appender<char>, long double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "fmt::v7::detail::error_handler::on_error(char const*)", referenced from:
      fmt::v7::basic_format_parse_context<char, fmt::v7::detail::error_handler>::on_error(char const*) in joinMap.cpp.o
      void fmt::v7::detail::handle_char_specs<char, fmt::v7::detail::char_specs_checker<fmt::v7::detail::error_handler> >(fmt::v7::basic_format_specs<char> const*, fmt::v7::detail::char_specs_checker<fmt::v7::detail::error_handler>&&) in joinMap.cpp.o
      fmt::v7::detail::float_specs fmt::v7::detail::parse_float_type_spec<fmt::v7::detail::error_handler&, char>(fmt::v7::basic_format_specs<char> const&, fmt::v7::detail::error_handler&) in joinMap.cpp.o
      void fmt::v7::detail::handle_cstring_type_spec<char, fmt::v7::detail::cstring_type_checker<fmt::v7::detail::error_handler> >(char, fmt::v7::detail::cstring_type_checker<fmt::v7::detail::error_handler>&&) in joinMap.cpp.o
      void fmt::v7::detail::check_string_type_spec<char, fmt::v7::detail::error_handler&>(char, fmt::v7::detail::error_handler&) in joinMap.cpp.o
      void fmt::v7::detail::check_pointer_type_spec<char, fmt::v7::detail::error_handler&>(char, fmt::v7::detail::error_handler&) in joinMap.cpp.o
      fmt::v7::detail::int_type_checker<fmt::v7::detail::error_handler>::on_error() in joinMap.cpp.o
      ...
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > fmt::v7::detail::grouping_impl<char>(fmt::v7::detail::locale_ref)", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > fmt::v7::detail::grouping<char>(fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "int fmt::v7::detail::snprintf_float<double>(double, int, fmt::v7::detail::float_specs, fmt::v7::detail::buffer<char>&)", referenced from:
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, float, 0>(fmt::v7::detail::buffer_appender<char>, float, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, double, 0>(fmt::v7::detail::buffer_appender<char>, double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "int fmt::v7::detail::snprintf_float<long double>(long double, int, fmt::v7::detail::float_specs, fmt::v7::detail::buffer<char>&)", referenced from:
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, long double, 0>(fmt::v7::detail::buffer_appender<char>, long double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "char fmt::v7::detail::decimal_point_impl<char>(fmt::v7::detail::locale_ref)", referenced from:
      char fmt::v7::detail::decimal_point<char>(fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "char fmt::v7::detail::thousands_sep_impl<char>(fmt::v7::detail::locale_ref)", referenced from:
      char fmt::v7::detail::thousands_sep<char>(fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "fmt::v7::vprint(fmt::v7::basic_string_view<char>, fmt::v7::format_args)", referenced from:
      void fmt::v7::print<Sophus::SO3Base<Sophus::SO3<double, 0> >::normalize()::'lambda'()::operator()() const::FMT_COMPILE_STRING, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> >, char>(Sophus::SO3<double, 0> const&, Sophus::SO3Base<Sophus::SO3<double, 0> >::normalize()::'lambda'()::operator()() const::FMT_COMPILE_STRING&&...) in joinMap.cpp.o
      void fmt::v7::print<char [2], char>(char const (&) [2]) in joinMap.cpp.o
  "std::__1::locale fmt::v7::detail::locale_ref::get<std::__1::locale>() const", referenced from:
      void fmt::v7::detail::format_value<char, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > >(fmt::v7::detail::buffer<char>&, Eigen::Transpose<Eigen::Matrix<double, 4, 1, 0, 4, 1> > const&, fmt::v7::detail::locale_ref) in joinMap.cpp.o
  "typeinfo for fmt::v7::format_error", referenced from:
      fmt::v7::detail::fill_t<char>::operator=(fmt::v7::basic_string_view<char>) in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned int>::on_error() in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned long long>::on_error() in joinMap.cpp.o
      fmt::v7::detail::int_writer<fmt::v7::detail::buffer_appender<char>, char, unsigned __int128>::on_error() in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, float, 0>(fmt::v7::detail::buffer_appender<char>, float, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, double, 0>(fmt::v7::detail::buffer_appender<char>, double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      fmt::v7::detail::buffer_appender<char> fmt::v7::detail::write<char, fmt::v7::detail::buffer_appender<char>, long double, 0>(fmt::v7::detail::buffer_appender<char>, long double, fmt::v7::basic_format_specs<char>, fmt::v7::detail::locale_ref) in joinMap.cpp.o
      ...
  "vtable for fmt::v7::format_error", referenced from:
      fmt::v7::format_error::format_error(char const*) in joinMap.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [joinMap] Error 1
make[1]: *** [CMakeFiles/joinMap.dir/all] Error 2
make: *** [all] Error 2

Solved it with the help of this:
https://github.com/fmtlib/fmt/issues/596#issuecomment-341617817

Specifically, I just put the following line in jointMap.cpp
#define FMT_HEADER_ONLY
